rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -355,120 +354,25 @@ func (s *RollupSyncService) getLocalChunksForBatch(chunkBlockRanges []*rawdb.Chu
-	}
-
-	commitBatchMeta.Version = version
-	commitBatchMeta.ChunkBlockRanges = ranges
-	return &commitBatchMeta, nil
-}
-
-// decodeBatchVersionAndChunkBlockRanges decodes version and chunks' block ranges in a batch based on the commit batch transaction's calldata.
-func (s *RollupSyncService) decodeBatchVersionAndChunkBlockRanges(txData []byte) (uint8, []*rawdb.ChunkBlockRange, error) {
-	const methodIDLength = 4
-	if len(txData) < methodIDLength {
-		return 0, nil, fmt.Errorf("transaction data is too short, length of tx data: %v, minimum length required: %v", len(txData), methodIDLength)
-	}
-
-	method, err := s.scrollChainABI.MethodById(txData[:methodIDLength])
-	if err != nil {
-		return 0, nil, fmt.Errorf("failed to get method by ID, ID: %v, err: %w", txData[:methodIDLength], err)
-	}
-
-	values, err := method.Inputs.Unpack(txData[methodIDLength:])
-	if err != nil {
-		return 0, nil, fmt.Errorf("failed to unpack transaction data using ABI, tx data: %v, err: %w", txData, err)
-	}
-
-	if method.Name == "commitBatch" {
-		type commitBatchArgs struct {
-			Version                uint8
-			ParentBatchHeader      []byte
-			Chunks                 [][]byte
-			SkippedL1MessageBitmap []byte
-		}
-
-		var args commitBatchArgs
-		if err = method.Inputs.Copy(&args, values); err != nil {
-			return 0, nil, fmt.Errorf("failed to decode calldata into commitBatch args, values: %+v, err: %w", values, err)
-		}
-
-		chunkRanges, err := decodeBlockRangesFromEncodedChunks(encoding.CodecVersion(args.Version), args.Chunks)
-		if err != nil {
-			return 0, nil, fmt.Errorf("failed to decode block ranges from encoded chunks, version: %v, chunks: %+v, err: %w", args.Version, args.Chunks, err)
-		}
-
-		return args.Version, chunkRanges, nil
-	} else if method.Name == "commitBatchWithBlobProof" {
-		type commitBatchWithBlobProofArgs struct {
-			Version                uint8
-			ParentBatchHeader      []byte
-			Chunks                 [][]byte
-			SkippedL1MessageBitmap []byte
-			BlobDataProof          []byte
-		}
-
-		var args commitBatchWithBlobProofArgs
-		if err = method.Inputs.Copy(&args, values); err != nil {
-			return 0, nil, fmt.Errorf("failed to decode calldata into commitBatchWithBlobProofArgs args, values: %+v, err: %w", values, err)
-		}
-
-		chunkRanges, err := decodeBlockRangesFromEncodedChunks(encoding.CodecVersion(args.Version), args.Chunks)
-		if err != nil {
-			return 0, nil, fmt.Errorf("failed to decode block ranges from encoded chunks, version: %v, chunks: %+v, err: %w", args.Version, args.Chunks, err)
-		}
-
-		return args.Version, chunkRanges, nil
+		return nil, fmt.Errorf("failed to decode block ranges from chunks, batch index: %v, err: %w", commitedBatch.BatchIndex(), err)
 	}
 
-	return 0, nil, fmt.Errorf("unexpected method name: %v", method.Name)
+	return &rawdb.CommittedBatchMeta{
+		Version:             uint8(commitedBatch.Version()),
+		ChunkBlockRanges:    chunkRanges,
+		BlobVersionedHashes: commitedBatch.BlobVersionedHashes(),
```

Comment:
this field is dangling (used in previous versions but not used after da-codec refactoring), should (and could) we remove it as well?